.github/workflows/update-changelog.yml

```diff
@@ -18,6 +18,36 @@ jobs:
           # is not out of date.
           ref: ${{ github.event.release.target_commitish }}
 
+      - name: Check if branch and release match
+        if: ${{ github.event.release.target_commitish != 'main' && github.event.release.target_commitish != 'master' }}
+        id: guard
+        run: |
+          NUMERIC_VERSION="${RELEASE_TAG_NAME#v}"
+          MAJOR_VERSION="${NUMERIC_VERSION%%.*}"
+          BRANCH_MAJOR_VERSION="${BRANCH%%.*}"
+
+          echo "MAJOR_VERSION=$(echo $MAJOR_VERSION)" >> $GITHUB_OUTPUT;
+          echo "BRANCH_MAJOR_VERSION=$(echo $BRANCH_MAJOR_VERSION)" >> $GITHUB_OUTPUT;
+
+          if [ "$MAJOR_VERSION" != "$BRANCH_MAJOR_VERSION" ]; then
+            echo "Mismatched versions! Aborting."
+            VERSION_MISMATCH='true';
+          else
+            echo "Versions match! Proceeding."
+            VERSION_MISMATCH='false';
+          fi
+          echo "VERSION_MISMATCH=$(echo $VERSION_MISMATCH)" >> $GITHUB_OUTPUT;
+        env:
+          BRANCH: ${{ github.event.release.target_commitish }}
+          RELEASE_TAG_NAME: ${{ github.event.release.tag_name }}
+
+      - name: Fail if branch and release tag do not match
+        if: ${{ steps.guard.outputs.VERSION_MISMATCH == 'true' }}
+        uses: actions/github-script@v6
+        with:
+          script: |
+              core.setFailed('Workflow failed. Release version does not match with selected target branch. Changelog not updated automatically.')
```

Comment:
Why do you have an extra step to fail the workflow? Can't you do this in the previous one?

Reply:
The first step runs bash. We could add [`exit 1;`](https://docs.github.com/en/actions/creating-actions/setting-exit-codes-for-actions#setting-a-failure-exit-code-in-a-docker-container-action) to the if-clause and abort the workflow there, but we then can't log a error message that is shown in the run summary. (One could `echo` something, but that would be hidden in the run logs)

Reply:
Gotcha 👍